### PR TITLE
rust 1.58.1

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by carlocab** [comment]
This will need the patch from #92548. The resources likely need updating too.

**Comment by branchvincent** [comment]
@carlocab does this look good now?

File: Formula/rust.rb [hunk 1]
```diff
@@ -130,3 +126,19 @@ def post_install
...
++++ Cargo.lock	2022-01-05 23:37:17.531044204 +0800
+@@ -658,9 +658,9 @@
+
+ [[package]]
+ name = "compiler_builtins"
+-version = "0.1.49"
++version = "0.1.55"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "20b1438ef42c655665a8ab2c1c6d605a305f031d38d9be689ddfef41a20f3aa2"
++checksum = "c9ac60765140c97aaf531dae151a287646b0805ec725805da9e2a3ee31cd501c"
```
[the remark this answers is not in the document]
Comment (branchvincent) [reply Formula/rust.rb]: thanks!

**Comment by SMillerDev** [comment]
```
   ==> /opt/homebrew/Cellar/rust/1.58.0/bin/cargo new hello_world --bin
       Created binary (application) `hello_world` package
     Compiling hello_world v0.1.0 (/private/tmp/rust-test-20220113-6831-1fcujcb/hello_world)
      Finished dev [unoptimized + debuginfo] target(s) in 0.12s
       Running `target/debug/hello_world`
  Error: rust: failed
  An exception occurred within a child process:
    Minitest::Assertion: Expected: "Hello, world!"
    Actual: nil
```

**Comment by carlocab** [comment]
I merged #92548, so this will need rebasing.

**Comment by pthariensflame** [comment]
This will require updating to 1.58.1 now.

File: Formula/rust.rb [hunk 2]
```diff
@@ -2,17 +2,16 @@ class Rust < Formula
...
+    url "https://static.rust-lang.org/dist/rustc-1.58.1-src.tar.gz"
+    sha256 "a839afdd3625d6f3f3c4c10b79813675d1775c460d14be1feaf33a6c829c07c7"
 
     # From https://github.com/rust-lang/rust/tree/#{version}/src/tools
     resource "cargo" do
       url "https://github.com/rust-lang/cargo.git",
-          tag:      "0.58",
-          revision: "b2e52d7cab0a286ee9fcc0c17510b1e72fcb53eb"
+          tag:      "0.59.0",
+          revision: "7f08ace4f1305de7f3b1b0e2f765911957226bd4"
```
Comment (branchvincent) [inline Formula/rust.rb:14]: technically the latest `cargo` is now one commit ahead of this, but it's untagged and only test changes: https://github.com/rust-lang/cargo/compare/0.59.0...rust-1.58.0

**Comment by carlocab** [comment]
Let's get this updated and merged: https://blog.rust-lang.org/2022/01/20/cve-2022-21658.html

Fine with skipping dependent tests for this one.

**Review by carlocab — DISMISSED** [review]

**Comment by carlocab** [comment]
Also, here's a fun thought: Rust binaries are statically linked, so everything built with a compiler older than Rust 1.58.1 is potentially susceptible to the previously mentioned vulnerability.

**Comment by pthariensflame** [comment]
> Also, here's a fun thought: Rust binaries are statically linked, so everything built with a compiler older than Rust 1.58.1 is potentially susceptible to the previously mentioned vulnerability.

It would have to use the relevant part of the stdlib (file deletion by path) and do so as a privileged process, but yes.

**Comment by carlocab** [comment]
Downloaded the bottle. Can't replicate the test failure locally.

**Review by carlocab — APPROVED** [review]
Not really sure why that fixed the test, but I'll take it.

**Review by SMillerDev — APPROVED** [review]

**Comment by carlocab** [comment]
No linkage with `libssh2`. We might not need that one anymore.

**Comment by carlocab** [comment]
> > Also, here's a fun thought: Rust binaries are statically linked, so everything built with a compiler older than Rust 1.58.1 is potentially susceptible to the previously mentioned vulnerability.
> 
> It would have to use the relevant part of the stdlib (file deletion by path) and do so as a privileged process, but yes.

Yes, this is what I meant by "potentially susceptible". Good thing we don't install things as `root`...

**Pull request closed without merging**